libcommon/strjoin.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * strjoin.c - concatenates strings from an array into a single one
+ *
+ * Mildly inspired by Go standard library's strings.Join function,
+ * but a little bit more rigorous to use since, well, this is C.
+ *
+ * Copyright (c) 2025 Luiz Antônio Rangel (takusuman)
+ *
+ * SPDX-Licence-Identifier: Zlib
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+char *strjoin(char *elems[], char *sep) {
+ 	size_t en = 0,
+	       joinlen = 0;
+	char *joinbuf = NULL,
+	     *joinbufp = NULL;
+
+	/*
+	 * Count strings in the array using 'NULL' as a sentinel.
+	 * Also already compute the length of each string in
+	 * the array if possible so we can save a loop later.
+	 */
+	for (; elems[en] != NULL; en++)
+		joinlen += strlen(elems[en]);
+	if (en == 1) return elems[0];
```

Comment:
Don't do this. Just think for a bit, the function usually returns a heap allocated string right? What if I use it with a constant `elems` array?, Like `{"bolsonaro", "preso"}`. It should return a heap allocated `"bolsonaro preso"`. Since I know this is heap allocated, I will do this: `s[2] = 's'; s[3] = 't'`. And then, after using `s`, I will run `free(s)`.

Now, let's say I passed `{"bolsonaro"}` to it. Your current code would return `"bolsonaro"`, a static string. Now, when I replace the characters, I will have issues since I shouldn't modify this sort of data. And even if there was no error for that, it would still fail when I use `free(s)`.

This little nearly useless optimization can make the function extremely difficult to use.

Reply:
I programmed it like I'd in Go.
Perhaps I should've used ``strdup()`` on ``elems[0]`` when returning it...

Reply:
Do you realize that you would be adding a branch in every execution of this function just to make it a very little more efficient if it has a single element?